### PR TITLE
[Yul] Add code generation for "return" and for loops.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -121,36 +122,101 @@ bool IRGeneratorForStatements::visit(Assignment const& _assignment)
...
-			m_context.variable(_binOp) <<
+			m_context.variableName(*returnParameters.front()) <<
 			" := " <<
-			m_utils.overflowCheckedUIntAddFunction(type->numBits()) <<
-			"(" <<
-			m_context.variable(_binOp.leftExpression()) <<
-			", " <<
-			m_context.variable(_binOp.rightExpression()) <<
-			")\n";
+			expressionAsType(*value, *returnParameters.front()->type()) <<
```
Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]:
```suggestion
			expressionAsType(*value, *types.at(0)) <<
```

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -190,24 +190,39 @@ bool IRGeneratorForStatements::visit(Return const& _return)
...
-			", " <<
-			m_context.variable(_binOp.rightExpression()) <<
-			")\n";
-	}
+			toCompactHexWithPrefix(commonType->literalValue(nullptr)) <<
+			"\n";
 	else
-		solUnimplementedAssert(false, "");
+	{
+		solUnimplementedAssert(_binOp.getOperator() == Token::Add, "");
```
Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: Already checked in the first if.

Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: Already checked in the first if.

**Comment by leonardoalt** [comment]
Tests are failing

**Comment by Marenz** [comment]
refs #6478

**Comment by chriseth** [comment]
This does not fix #6531 - it implements Solidity's `return` statement, but #6531 is about adding a `return` statement to yul.

**Comment by Marenz** [comment]
Oopps. My bad

**Comment by chriseth** [comment]
Fixed failure and added a test.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**